### PR TITLE
add agent point

### PR DESCRIPTION
**Pull request by mat1e**
Link to #150 

Add `agent_point` with : 
* `agent_name` : name of the agent
* `agent_label` : label of the agent

### CODE REVIEW

**Comment by asimell** [comment]
I feel like adding a completely new measurement is a slight overkill for effectively 1 new metric (`build_agent_name` is already available in `jenkins_data`). What is the benefit of this new measurement compared to just adding the agent label as a new metric for `jenkins_data` as well? And is there any other relevant data one could collect from the agent other than the name and label? I'm having a hard time grasping what is the overall use case of having such a tiny measurement.

**Comment by mat1e** [comment]
> 

In agent_datas, you can find all agents called by the pipeline. I tried to get others informations like the time the build use the agent  but I didn't find a way to get it yet. Because the agents are already removed (in case of cloud agents). For now the only benefit is to have all agents names and label used by a pipeline.

**Comment by asimell** [comment]
Could you also add unit tests to demonstrate this behaviour? Currently the tests only test with one agent, which is exactly the same behaviour as the `build_agent_name` in `jenkins_data`. What this looks like to me is that you're writing multiple points on the same timestamp, which in InfluxDB results into overwriting data. For example, with `RobotFrameworkPointGenerator` we need to manually set the timestamp, so that data is not overwritten and I think something similar is needed here.

**Comment by mat1e** [comment]
> Could you also add unit tests to demonstrate this behaviour? Currently the tests only test with one agent, which is exactly the same behaviour as the `build_agent_name` in `jenkins_data`. What this looks like to me is that you're writing multiple points on the same timestamp, which in InfluxDB results into overwriting data. For example, with `RobotFrameworkPointGenerator` we need to manually set the timestamp, so that data is not overwritten and I think something similar is needed here.

You're right, we just tested and datas are overwrited... I will do the modifications.

**Comment by asimell** [comment]
The plugin contains a class called `TimeGenerator`, which you can use to iterate 1 nanosecond onwards to get a  different timestamp, so the easiest way to avoid data overwrite would be to do something like this
```
TimeGenerator generator = new TimeGenerator(timestamp);
for (...) {
    long time = generator.next();
    Point point = BuildPoint("measurement", customPrefix, build, time)
   ...;
}
```

**Comment by mat1e** [comment]
> The plugin contains a class called `TimeGenerator`, which you can use to iterate 1 nanosecond onwards to get a different timestamp, so the easiest way to avoid data overwrite would be to do something like this
> 
> ```
> TimeGenerator generator = new TimeGenerator(timestamp);
> for (...) {
>     long time = generator.next();
>     Point point = BuildPoint("measurement", customPrefix, build, time)
>    ...;
> }
> ```

It is not good with an unique_id ? I don't want to corrupt the datas...

**Comment by asimell** [comment]
A unique ID tag works as well. Just pointing out a solution for not reinventing the wheel, but you had already done your own solution. 😊

File: doc/available_metrics.md [hunk 1]
```diff
@@ -112,8 +112,11 @@ Tags specific for this measurement:
 | sqale_index | float | Technical Debt | 2.4 |
 | sqale_debt_ratio | float | Technical Debt Ratio | 2.4 |
 
-
-
+#### `agent_data`
```
Comment (asimell) [inline doc/available_metrics.md]:
```suggestion
#### `agent_data` (since 3.4)
```

**Comment by HabibMAALEM** [comment]
> A unique ID tag works as well. Just pointing out a solution for not reinventing the wheel, but you had already done your own solution. 😊

A unique_id tag such an incremental number is suitable for this case (especially for performance), example : 

`unique_id=1`
`unique_id=2`

**Review by asimell — APPROVED** [review]
LGTM

**Pull request merged**